spring-cloud-dataflow-registry/src/main/java/org/springframework/cloud/dataflow/registry/AppRegistry.java

```diff
@@ -235,4 +238,12 @@ else if (!StringUtils.hasText(uri.getSchemeSpecificPart())) {
 		}
 		return uri;
 	}
+
+	private void validateApplicationUri(URI uri){
+		if (uri.getScheme().toLowerCase().equals("file")){
+			File file = new File(uri);
+			Assert.isTrue(file.exists() && !file.isDirectory() && file.getName().toLowerCase().endsWith(".jar"),
```

Comment:
Hmm, that changes things. Wasn't aware of that. Perhaps I need to revisit the approach all together to at least ensure that that user has a better way of being notified that the final stream can not be deployed due to. . . .